documentation/examples/remote_storage/remote_storage_adapter/influxdb/client.go

```diff
@@ -77,6 +80,22 @@ func tagsFromMetric(m model.Metric) map[string]string {
 	return tags
 }
 
+// ensureDatabase ensures the existence of the database.
+func (c *Client) ensureDatabase() {
+	q := influx.Query{
+		Command: fmt.Sprintf(`CREATE DATABASE %s`, c.database),
+	}
+	tk := time.NewTicker(5 * time.Second)
+	defer tk.Stop()
+	for range tk.C {
+		_, err := c.client.Query(q)
+		if err == nil {
+			return
+		}
+		level.Warn(c.logger).Log("msg", "failed to create database", "err", err)
+	}
+}
```

Comment:
shouldn't we also have some fail safe with time.After our a counter to give up after many failed attempts or retries?

Reply:
The timeout can be tricky in a cloud environment.

For our use case, we run both remote storage adaptor and influxdb in kubernetes clusters (including some temporary testing clusters), it's hard to decide how long it should take for the influxdb to be scheduled. That's why I originally made ensureDatabase() non-blocking,  to make the remote storage adaptor don't wait influxdb. The database will be eventually created when influxdb is up.

Let me know if you have some suggestions.

Reply:
I see the logic now. 
Is there no way to check the error type when writing so that you can create the DB at that point.
`if err.Type==...databaseNoExist { create database.}`

Reply:
Just had a quick look at the influx  client code and doesn't  seem possible to detect the error type apart from matching a string which will brake quite easy for different server versions.
My only other idea so to open an issue upstream and explain them the use case , they might have some other ideas or might be willing to accept a PR to allow detecting the `no DB` errors from the client.

Reply:
Just had a look at telegraf (the official product of influxdata):
https://github.com/influxdata/telegraf/blob/master/plugins/outputs/influxdb/influxdb.go#L195,
it is actually using string matching to detect "database not found"  error.
I could follow that if you think it's reasonable.

Reply:
Yeah that will work , but would be very easy to break. 
Would you mind opening an issue with upstream and see if they are willing to add something better to detect error types from the client.
That would be very easy to add and it is worth asking I think.

Reply:
or even comparing the string to some constant would be fine. Just that the constant needs to come from the influx's client package so that if they update the package they would also update the constant.

Reply:
opened https://github.com/influxdata/telegraf/issues/3820

Reply:
Thanks :+1: